internal/provider/user_data_source.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &userDataSource{}
+
+func NewuserDataSource() datasource.DataSource {
+	return &userDataSource{}
+}
+
+// userDataSource defines the data source implementation.
+type userDataSource struct {
+	apiToken string
+}
+
+// userDataSourceModel describes the data source data model.
+type userDataSourceModel struct {
+	Id    types.String `tfsdk:"id"`
+	Email types.String `tfsdk:"email"`
+}
+
+func (d *userDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user"
+}
+
+func (d *userDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "User",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The ID of this user.",
+			},
+			"email": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The email of this user.",
+			},
+		},
+	}
+}
+
+func (d *userDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	apiToken, ok := req.ProviderData.(string)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected string, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.apiToken = apiToken
+}
+
+func (d *userDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state userDataSourceModel
+
+	///////////////////// API Call
+
+	// TODO: update to retrieve ID from Terraform configuration.
+	url := fmt.Sprintf(BASE_URL+"/"+USER_BY_ID_URL, "95997")
+	httpReq, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		fmt.Println("Error creating request:", err)
+		return
+	}
+
+	httpReq.Header.Set("Authorization", "Bearer "+d.apiToken)
```

Comment:
Ah, I see. I wonder if there is some way we can abstract the authn/authz out of the individual `Read` methods, but I do not know what the best practices here are.